### PR TITLE
created file  checkpoints.go with all function and structs

### PR DESCRIPTION
**Pull request by DarkLord017**

### CODE REVIEW

**Comment by DarkLord017** [comment]
Implemented checkpoints.go with the structs and functions given , tests not written 
@gerceboss

File: config/checkpoints/checkpoints.go [hunk 1]
```diff
@@ -1,64 +1,366 @@
...
-func deserialize_slot() {}
+// @param url: string - url to fetch
+// @return *http.Response, error
+// Fetches the response from the given url
+func get(url string) (*http.Response, error) {
+	var resp *http.Response
+
+	// Retry with default settings
+	err := retry.Do(
+		func() error {
```
Comment (gerceboss) [inline config/checkpoints/checkpoints.go:65]: How did you handle `BackOffSettings::default()` and also add comments in the structs as well, what do they represent (can refer Helios)

Comment (DarkLord017) [reply config/checkpoints/checkpoints.go:65]: image
the library had default of there own so i just wrote the function
I will add the comments sir

Comment (DarkLord017) [reply config/checkpoints/checkpoints.go:83]: Sir i have added the comments please check

Comment (DarkLord017) [reply config/checkpoints/checkpoints.go:83]: @gerceboss

**Review by star-gazer111 — APPROVED** [review]
LGTM.

**Pull request merged**